google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -1285,6 +1318,49 @@ public void testUpdateDatasetWithSelectedFields() {
     assertTrue(dataset.delete());
   }
 
+  @Test
+  public void testUpdateDatabaseWithAccessPolicyVersion() throws IOException {
+    String accessPolicyDataset = RemoteBigQueryHelper.generateDatasetName();
+    ServiceAccountCredentials credentials =
+        (ServiceAccountCredentials) GoogleCredentials.getApplicationDefault();
+    Dataset dataset =
+        bigquery.create(
+            DatasetInfo.newBuilder(accessPolicyDataset)
+                .setDescription("Some Description")
+                .setLabels(Collections.singletonMap("a", "b"))
+                .build());
+    assertThat(dataset).isNotNull();
+
+    User user = new User(credentials.getClientEmail());
+    Acl.Role role = Acl.Role.WRITER;
+    Acl.Expr condition =
+        new Expr(
+            "request.time < timestamp('2030-01-01T00:00:00Z')",
+            "test condition",
+            "requests before the year 2030",
+            "location");
+    Acl acl = Acl.of(user, role, condition);
+    List<Acl> acls = new ArrayList<>();
+    acls.addAll(dataset.getAcl());
+    acls.add(acl);
+
+    DatasetOption datasetOption = DatasetOption.accessPolicyVersion(3);
+    Dataset updatedDataset =
+        bigquery.update(
+            dataset
+                .toBuilder()
+                .setDescription("Updated Description")
+                .setLabels(null)
+                .setAcl(acls)
+                .build(),
+            datasetOption);
+    assertNotNull(updatedDataset);
+    assertEquals(updatedDataset.getDescription(), "Updated Description");
+    assertThat(updatedDataset.getLabels().isEmpty());
```

Comment:
Currently, is there a way to validate the set the expr value?
I see a "condition" parameter in the REST discovery but I'm not sure if it is populated.

Reply:
Tests have been updated to verify the remote datasets' conditions. The position of the returned remote dataset Acl containing the condition is not consistent, so the test has to iterate through all of the Acl objects to find the right one.